src/python/pants/jvm/compile.py

```diff
@@ -112,13 +113,14 @@ def for_targets(
                 continue
             elif classification == _ClasspathEntryRequestClassification.COMPATIBLE:
                 compatible.append(impl)
+                compatible_root_only.append(impl.root_only)
             elif classification == _ClasspathEntryRequestClassification.PARTIAL:
                 partial.append(impl)
             elif classification == _ClasspathEntryRequestClassification.CONSUME_ONLY:
                 consume_only.append(impl)
 
         if len(compatible) == 1:
-            if not root and impl.root_only:
+            if not root and compatible_root_only[0]:
```

Comment:
```suggestion
            if not root and compatible[0].root_only:
```
Since there is exactly one entry in the list, it can be inspected directly.

Reply:
Fixed.